xarray/backends/api.py

```diff
@@ -233,7 +233,7 @@ def _get_mtime(filename_or_obj):
 
 def _protect_dataset_variables_inplace(dataset, cache):
     for name, variable in dataset.variables.items():
-        if name not in variable.dims:
+        if name not in dataset.indexes:
```

Comment:
```suggestion
        if name not in dataset._indexes:
```

This saves the creation of an `Indexes` object that is not needed here (we only check for the keys).